### PR TITLE
[MDB IGNORE] The chair desouling and restructuring update (alongside some other changes)

### PR DESCRIPTION
**Pull request by PestoVerde322**
## About The Pull Request

This PR finally sort out and resprite all the chairs, alongside adding some new one (like the plastic chair or the foldable chair).

now the chairs are divided in two different categories:
-Normal, or chairs without any armstand
-Fancy, or chairs that may or may not have an armstand (like armchairs and sofas, but also benches)

This PR also adds Concave sofas, the option to colour up (for now) the old sofa and armchair with crayons.

## Why It's Good For The Game

More artistic cohesion, more detailing options, overall a more logical structure for chair and chair types.

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

_**For now i'll post this screenshot, further in game sprites will follow**_

chairs2pointOH

</details>

## Changelog
:cl:
add: Added the plastic chair, steel bench, folding chair, chair with arms (aka fancy chair), corporate armchair, dirty mattress
add: Added king sized variants of all the beds with their respective bedsheets, remember, you're constantly under observation by the administration team
add: Added concave corners for all the sofas.
imageadd: added brand new icons for chairs, beds, bedsheets and some benches, including renewed in-hand icons
code: Restructured how chairs are really sorted out
/:cl:

### CODE REVIEW

**Comment by giacomoh** [comment]
YES!! chairs!!

**Comment by PestoVerde322** [comment]
Added "Needs Testmerge due to me not being 100% sure i updated all the maps with the current chair hierarchy

**Comment by PowerfulBacon** [comment]
Double beds? Oh no

**Comment by itsmeow** [comment]
+20 for double beds

File: code/game/objects/structures/beds_chairs/bench.dm [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/obj/structure/chair/fancy/bench
+	name = "bench"
+	desc = "You sit in this. Either by will or force, but not alone maybe."
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/bench.dm]: but maybe not alone

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 2]
```diff
@@ -88,7 +91,7 @@
 	else
 		return ..()
 
-/obj/structure/chair/attack_tk(mob/user)
+/obj/structure/chaiar/attack_tk(mob/user)
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/chair.dm]: chaiar?

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 3]
```diff
@@ -120,6 +123,28 @@
 	handle_rotation(newdir)
 
 // Chair types
+
+/obj/structure/chair/old
+	name = "strange chair"
+	desc = "You sit in this. Either by will or force. Looks REALLY uncomfortable."
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/chair.dm]: VERY uncomfortable

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 4]
```diff
@@ -135,251 +160,111 @@
...
-	desc = "It looks comfy."
-	icon_state = "comfychair"
-	color = rgb(255,255,255)
-	resistance_flags = FLAMMABLE
-	max_integrity = 70
-	buildstackamount = 2
-	item_chair = null
+/obj/structure/chair/fancy //base for any chair with armrests
+	name = "fancy chair"
+	desc = "Giving you the feel of importance by just having armrests."
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/chair.dm]: It gives you the feel of importance, just by having armrests.

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 5]
```diff
@@ -66,14 +69,14 @@
 	qdel(src)
 
 /obj/structure/chair/ratvar_act()
-	var/obj/structure/chair/brass/B = new(get_turf(src))
+	var/obj/structure/chair/fancy/brass/B = new(get_turf(src))
 	B.setDir(dir)
 	qdel(src)
 
-/obj/structure/chair/attackby(obj/item/W, mob/user, params)
+/obj/structure/chair/attacked_by(obj/item/W, mob/user, params)
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/chair.dm]: I'm not exactly sure if we use attacked_by? do both work on bee?

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 6]
```diff
@@ -135,251 +160,111 @@
...
 	desc = "A comfortable, secure seat. It has a more sturdy looking buckling system for smoother flights."
 	icon_state = "shuttle_chair"
 
-/obj/structure/chair/comfy/shuttle/GetArmrest()
-	return mutable_appearance('icons/obj/chairs.dmi', "shuttle_chair_armrest")
-
-/obj/structure/chair/office
+/obj/structure/chair/fancy/plastic
+	name = "plastic chair"
+	desc = "If you want it, then you'll have to take it."
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/chair.dm:240]: I don't get this, is this a reference?

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 7]
```diff
@@ -135,251 +160,111 @@
...
+	handle_layer()
+	anchored = FALSE
 
-/obj/item/chair/wood/wings
-	icon_state = "wooden_chair_wings_toppled"
-	origin_type = /obj/structure/chair/wood/wings
+/obj/structure/chair/fancy/plastic/proc/snap_check(mob/living/M)
+	if (M.nutrition >= NUTRITION_LEVEL_FAT) //you are so fat
+		to_chat(M, "<span class='warning'>The chair begins to pop and crack, you're too heavy!</span>")
+		if(do_after(M, 6 SECONDS, progress = FALSE))
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/chair.dm:263]: can this be cancelled by getting up from the chair?

File: code/game/objects/structures/beds_chairs/sofa.dm [hunk 8]
```diff
@@ -1,67 +1,69 @@
-/obj/structure/chair/sofa
-	name = "old ratty sofa"
-	icon_state = "sofamiddle"
-	icon = 'icons/obj/sofa.dmi'
+/obj/structure/chair/fancy/sofa //like it's brother fancy chair, this is the father of all sofas
+	name = "old father sofa"
+	desc = "Now extint, this kind of sofa shouldn't even exists anymore, if you see this wild specimen, contact your local Nanotransen Anti-couch surfer department."
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/sofa.dm]: Now extinct, this kind of sofa shouldn't even exist anymore. If you see this wild specimen, contact your station's Anti-Couch Surfing Department.

File: code/game/objects/structures/beds_chairs/sofa.dm [hunk 9]
```diff
@@ -1,67 +1,69 @@
...
-/obj/structure/chair/sofa/Initialize(mapload)
-	armrest = mutable_appearance(icon, "[icon_state]_armrest", ABOVE_MOB_LAYER)
-	return ..()
-/obj/structure/chair/sofa/post_buckle_mob(mob/living/M)
-	. = ..()
-	update_armrest()
+/obj/structure/chair/fancy/sofa/old
+	color = rgb(141,70,0)
+	name = "old sofa"
+	desc = "An old design, but still does the job of being a sofa."
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/sofa.dm]: "but it still"

File: code/game/objects/structures/beds_chairs/sofa.dm [hunk 10]
```diff
@@ -1,67 +1,69 @@
...
-/obj/structure/chair/sofa/post_unbuckle_mob()
-	. = ..()
-	update_armrest()
+/obj/structure/chair/fancy/sofa/old/right
+	icon_state = "sofa_end_right"
 
-/obj/structure/chair/sofa/corner/handle_layer() //only the armrest/back of this chair should cover the mob.
+/obj/structure/chair/fancy/sofa/old/corner
+	name = "impossible old sofa corner"
+	desc = "this kind of sofa shouldn't even exists at all, if you see this non-euclidan specimen, contact your local Nanotransen Anti-couch surfer department."
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/sofa.dm]: This kind of sofa shouldn't even exist at all. If you see this non-euclidean specimen, contact your station's Anti-Couch Surfing Department.

File: code/game/objects/structures/beds_chairs/sofa.dm [hunk 11]
```diff
@@ -1,67 +1,69 @@
...
+
+/obj/structure/chair/fancy/sofa/old/corner/handle_layer() //only the armrest/back of this chair should cover the mob.
 	return
 
-/obj/structure/chair/sofa/left
-	icon_state = "sofaend_left"
+/obj/structure/chair/fancy/sofa/old/corner/concave
+	icon_state = "sofa_corner_in"
+	name = "old sofa"
+	desc = "this kind of sofa definitely exists and there's nothing wrong with it, this one is concave."
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/sofa.dm]: Just completely redo this, referencing other sofa descriptions in another one makes no sense

File: code/game/objects/structures/beds_chairs/sofa.dm [hunk 12]
```diff
@@ -1,67 +1,69 @@
...
 
-/obj/structure/chair/sofa/right
-	icon_state = "sofaend_right"
+/obj/structure/chair/fancy/sofa/old/corner/convex
+	icon_state = "sofa_corner_out"
+	name = "old sofa"
+	desc = "this kind of sofa definitely exists and there's nothing wrong with it, this one is convex."
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/sofa.dm]: same for this as the above one

File: code/game/objects/structures/beds_chairs/sofa.dm [hunk 13]
```diff
@@ -1,67 +1,69 @@
...
-/obj/structure/chair/sofa/corp/right
-	icon_state = "corp_sofaend_right"
+/obj/structure/chair/fancy/sofa/corp/right
+	icon_state = "corp_sofa_end_right"
 
-/obj/structure/chair/sofa/corp/corner
-	icon_state = "corp_sofacorner"
+/obj/structure/chair/fancy/sofa/corp/corner
+	name = "impossible corporate sofa corner"
+	desc = "this kind of sofa shouldn't even exists at all, if you see this non-euclidan specimen, contact your local Nanotransen Anti-couch surfer department."
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/sofa.dm]: see other description

File: code/game/objects/structures/beds_chairs/sofa.dm [hunk 14]
```diff
@@ -1,67 +1,69 @@
...
-	buildstackamount = 3
+/obj/structure/chair/fancy/sofa/corp/corner/handle_layer()
+	return
 
-/obj/structure/chair/sofa/bamboo/left
-	icon_state = "bamboo_sofaend_left"
+/obj/structure/chair/fancy/sofa/corp/corner/concave
+	icon_state = "corp_sofa_corner_in"
+	name = "corporate sofa"
+	desc = "this kind of professional sofa definitely exists and there's nothing wrong with it, this one is concave."
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/sofa.dm]: same as others

File: code/game/objects/structures/beds_chairs/sofa.dm [hunk 15]
```diff
@@ -1,67 +1,69 @@
...
 
-/obj/structure/chair/sofa/bamboo/right
-	icon_state = "bamboo_sofaend_right"
+/obj/structure/chair/fancy/sofa/corp/corner/convex
+	icon_state = "corp_sofa_corner_out"
+	name = "corporate sofa"
+	desc = "this kind of professional sofa definitely exists and there's nothing wrong with it, this one is convex."
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/sofa.dm]: same as others

File: code/game/objects/structures/electricchair.dm [hunk 16]
```diff
@@ -8,17 +8,16 @@
...
 	. = ..()
-	add_overlay(mutable_appearance('icons/obj/chairs.dmi', "echair_over", MOB_LAYER + 1))
+	add_overlay(mutable_appearance('icons/obj/beds_chairs/chairs.dmi', "echair_over", MOB_LAYER + 1))
 
 /obj/structure/chair/e_chair/attackby(obj/item/W, mob/user, params)
 	if(W.tool_behaviour == TOOL_WRENCH)
 		var/obj/structure/chair/C = new /obj/structure/chair(loc)
 		W.play_tool_sound(src)
 		C.setDir(dir)
-		part.forceMove(loc)
```
Comment (itsmeow) [inline code/game/objects/structures/electricchair.dm:18]: was this part of the parent function? why was this removed?

File: code/modules/cargo/bounties/assistant.dm [hunk 17]
```diff
@@ -122,7 +122,7 @@
 	description = "Commander Pat is unhappy with his chair. He claims it hurts his back. Ship some alternatives out to humor him."
 	reward = 1500
 	required_count = 5
-	wanted_types = list(/obj/structure/chair/comfy)
+	wanted_types = list(/obj/structure/chair/fancy/comfy/)
```
Comment (itsmeow) [inline code/modules/cargo/bounties/assistant.dm]: hanging `/`

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 18]
```diff
@@ -425,31 +310,89 @@
...
-/obj/structure/chair/bronze/Moved()
+/obj/structure/chair/fancy/brass/bronze/Moved()
 	. = ..()
 	if(has_gravity())
 		playsound(src, 'sound/machines/clockcult/integration_cog_install.ogg', 50, TRUE)
 
-/obj/structure/chair/mime
-	name = "invisible chair"
-	desc = "The mime needs to sit down and shut up."
+/obj/structure/chair/office
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/chair.dm:327]: no name or desc?

File: code/game/objects/structures/beds_chairs/bed.dm [hunk 19]
```diff
@@ -217,3 +218,39 @@
...
+		goldilocks = null
+
+/obj/structure/bed/double/maint
+	name = "double dirty mattress"
+	desc = "An old grubby king sized mattress. You really try to not think about what could be the cause of those stains."
+	icon_state = "dirty_mattress_double"
+
+/obj/structure/bed/double/alien
+	name = "double resting contraption"
+	desc = "This looks similar to contraptions from Earth. Could aliens be stealing our technology?"
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/bed.dm]: maybe override Examine and check if they're an abductor for some fun interaction

File: code/datums/components/crafting/recipes.dm [hunk 20]
```diff
@@ -1025,3 +1025,23 @@
...
+	name = "Foldable Chair"
+	result = /obj/item/chair/foldable
+	time = 40
+	reqs = list(/obj/item/stack/rods = 2,
+				/obj/item/stack/sheet/plastic = 1
+				)
+	tools = list(TOOL_WRENCH, TOOL_WIRECUTTER)
+	category = CAT_MISC
+
+/datum/crafting_recipe/chairfancy
```
Comment (itsmeow) [inline code/datums/components/crafting/recipes.dm]: rename to chair_fancy

File: code/datums/components/crafting/recipes.dm [hunk 21]
```diff
@@ -1025,3 +1025,23 @@
...
+	name = "fancy Chair"
```
Comment (itsmeow) [inline code/datums/components/crafting/recipes.dm]: capitalize fancy

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Comment by itsmeow** [comment]
Map compilation errors

https://github.com/BeeStation/BeeStation-Hornet/runs/7957367075?check_suite_focus=true#step:5:73

File: code/game/objects/structures/beds_chairs/chair.dm (hunk 6 above)
Comment (PestoVerde322) [reply code/game/objects/structures/beds_chairs/chair.dm:240]: Yep, it's from devil may cry 5.

File: code/game/objects/structures/beds_chairs/bed.dm (hunk 19 above)
Comment (PestoVerde322) [reply code/game/objects/structures/beds_chairs/bed.dm]: Not 100% sure on how to do it.

File: beestation.dme [hunk 22]
```diff
@@ -3544,4 +3541,8 @@
 #include "interface\interface.dm"
 #include "interface\stylesheet.dm"
 #include "interface\skin.dmf"
+#include "code\game\objects\items\stacks\tiles\tile_reskinning.dm"
```
Comment (itsmeow) [inline beestation.dme]: please put these where they were, it gets auto-ordered by local VScode and you're gonna end up with another `cult_ritual_item.dm` situation

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: beestation.dme (hunk 22 above)
Comment (PestoVerde322) [reply beestation.dme]: honestly, i don't even know how the hell this happened, i repasted them where they where supposed to be and yet here we are, thanks VSC.

File: code/__DEFINES/dye_keys.dm [hunk 23]
```diff
@@ -4,6 +4,7 @@
 #define DYE_REGISTRY_SNEAKERS "sneakers"
 #define DYE_REGISTRY_FANNYPACK "fannypack"
 #define DYE_REGISTRY_BEDSHEET "bedsheet"
+#define DYE_REGISTRY_DOUBLE_BEDSHEET "double bedsheet"
```
Comment (itsmeow) [inline code/__DEFINES/dye_keys.dm]: this should be `double_bedsheet` if it can be

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 24]
```diff
@@ -17,92 +17,105 @@
...
+		)), \
+	new /datum/stack_recipe_list("corporate benches", list( \
+		new /datum/stack_recipe("corporate bench (middle)",				/obj/structure/chair/fancy/bench/corporate, 1, one_per_turf = TRUE, on_floor = TRUE), \
+		new /datum/stack_recipe("corporate bench (left)",				/obj/structure/chair/fancy/bench/corporate/left, 1, one_per_turf = TRUE, on_floor = TRUE), \
+		new /datum/stack_recipe("corporate bench (right)",				/obj/structure/chair/fancy/bench/corporate/right, 1, one_per_turf = TRUE, on_floor = TRUE), \
 		)),
 	null, \
-	new/datum/stack_recipe("rack parts", /obj/item/rack_parts), \
-	new/datum/stack_recipe("closet", /obj/structure/closet, 2, time = 15, one_per_turf = TRUE, on_floor = TRUE), \
+	new/datum/stack_recipe("rack parts",								/obj/item/rack_parts), \
```
Comment (itsmeow) [inline code/game/objects/items/stacks/sheets/sheet_types.dm:63]: not a huge fan of needlessly reformatting an entire file, it makes it hard to see a diff and just makes a ton of possible merge conflicts

File: code/game/objects/structures/beds_chairs/bed.dm [hunk 25]
```diff
@@ -217,3 +218,39 @@
...
+
+//Double Beds, for luxurious sleeping, i.e. the captain and maybe heads - if people use this for ERP, send them to skyrat, or worse, acacia
+/obj/structure/bed/double
+	name = "double bed"
+	desc = "A luxurious double bed, for those too important for small dreams."
+	icon_state = "bed_double"
+	buildstackamount = 4
+	max_buckled_mobs = 2
+	///The mob who buckled to this bed second, to avoid other mobs getting pixel-shifted before they unbuckles.
+	var/mob/living/goldilocks
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/bed.dm:241]: possible hard-del.

Register a QDELETING signal - on qdel set to null and reset the pixel height

File: code/game/objects/structures/beds_chairs/bed.dm (hunk 19 above)
Comment (itsmeow) [reply code/game/objects/structures/beds_chairs/bed.dm]:
```
/obj/structure/bed/double/alien/Examine(mob/user)
    . = ..()
    if(istype(user, /mob/living/carbon/human/species/abductor))
        . += "It looks like a regular bed."
```

Would need some way to remove the parent description, also this is just a very rough code and won't actually work

File: code/game/objects/structures/beds_chairs/bench.dm [hunk 26]
```diff
@@ -0,0 +1,67 @@
...
+	desc = "A makeshift bench with a rustic aesthetic."
+	icon_state = "bamboo_sofamiddle"
+	resistance_flags = FLAMMABLE
+	max_integrity = 60
+	buildstacktype = /obj/item/stack/sheet/mineral/bamboo
+	buildstackamount = 3
+
+/obj/structure/chair/fancy/bench/bamboo/left
+	icon_state = "bamboo_sofaend_left"
+/obj/structure/chair/fancy/bench/bamboo/right
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/bench.dm:46]: missing spacing between the two

File: code/game/objects/structures/beds_chairs/bed.dm (hunk 25 above)
Comment (itsmeow) [reply code/game/objects/structures/beds_chairs/bed.dm:241]: could also use a weakref

https://hackmd.io/@PowerfulBacon/guide_to_harddels

File: code/game/objects/structures/beds_chairs/chair.dm (hunk 18 above)
Comment (itsmeow) [reply code/game/objects/structures/beds_chairs/chair.dm:327]: still no name

File: code/game/objects/structures/bedsheet_bin.dm [hunk 27]
```diff
@@ -287,6 +287,133 @@ LINEN BINS
...
+				/obj/item/bedsheet/brown/double,
+				/obj/item/bedsheet/black/double))
+		if("Special")
+			type = pick(list(/obj/item/bedsheet/patriot/double,
+				/obj/item/bedsheet/rainbow/double,
+				/obj/item/bedsheet/ian/double,
+				/obj/item/bedsheet/cosmos/double,
+				/obj/item/bedsheet/nanotrasen/double))
+	new type(loc)
+	return INITIALIZE_HINT_QDEL
```
Comment (itsmeow) [inline code/game/objects/structures/bedsheet_bin.dm:415]: What's this for?

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Comment by itsmeow** [comment]
@PestoVerde322 Please mark my reviews as Resolved when you fix them

File: code/game/objects/structures/bedsheet_bin.dm (hunk 27 above)
Comment (itsmeow) [reply code/game/objects/structures/bedsheet_bin.dm:415]: picks random bedsheet then deletes this object*

File: code/game/objects/structures/beds_chairs/bed.dm (hunk 25 above)
Comment (PestoVerde322) [reply code/game/objects/structures/beds_chairs/bed.dm:241]: leaving this open because not sure if i fixed it?

File: code/game/objects/structures/beds_chairs/bed.dm [hunk 28]
```diff
@@ -265,4 +273,4 @@
 /obj/structure/bed/double/alien/examine(mob/user)
 	. = ..()
 	if(isabductor(user))
-		. += "<span class='abductor'>Fairly sure we absoluteley steal that technology...Why did we steal this again?</span>"
+		. += "<span class='abductor'>Fairly sure we absoluteley stole that technology...Why did we stole this again?</span>"
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/bed.dm]: Copy paste this please.
`Fairly sure we absolutely stole that technology... Why did we steal this again?`
`absoluteley` is wrong, second one is `steal`

File: code/game/objects/structures/beds_chairs/bed.dm [hunk 29]
```diff
@@ -243,22 +243,20 @@
...
 		goldilocks = null
-		UnregisterSignal(buckled_mobs, COMSIG_PARENT_QDELETING)
 
 //Called when the signal is raised, removes the reference
 //preventing the hard delete.
-/obj/structure/bed/double/proc/buckled_mobs_deleted(datum/source, force)
-    UnregisterSignal(buckled_mobs, COMSIG_PARENT_QDELETING)
-    goldilocks = null
+/obj/structure/bed/double/proc/goldilocks_deleted(datum/source, force)
+	UnregisterSignal(goldilocks, COMSIG_PARENT_QDELETING)
```
Comment (itsmeow) [inline code/game/objects/structures/beds_chairs/bed.dm:258]: this uses spaces. bad

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Review by itsmeow — APPROVED** [review]

**Review by PowerfulBacon — APPROVED** [review]
